eth2near/eth2near-block-relay-rs/src/beacon_rpc_client.rs

```diff
@@ -1,27 +1,94 @@
-use serde_json::Value;
+use crate::execution_block_proof::ExecutionBlockProof;
+use eth_types::eth2::BeaconBlockHeader;
+use eth_types::eth2::FinalizedHeaderUpdate;
+use eth_types::eth2::HeaderUpdate;
+use eth_types::eth2::LightClientUpdate;
+use eth_types::eth2::Slot;
+use eth_types::eth2::SyncCommittee;
+use eth_types::eth2::SyncCommitteeUpdate;
+use eth_types::eth2::SyncAggregate;
+use serde_json::{json, Value};
+use reqwest::blocking::Client;
 use std::error::Error;
-use types::BeaconBlockBody;
-use types::BeaconBlockHeader;
+use std::fmt;
+use std::fmt::Display;
+use std::string::String;
+use std::time::Duration;
+use eth_types::H256;
+use types::{BeaconBlockBody, BeaconState};
 use types::MainnetEthSpec;
 
-async fn get_json_from_raw_request(url: &str) -> Result<std::string::String, reqwest::Error> {
-    reqwest::get(url).await?.text().await
+#[derive(Debug)]
+pub struct SignatureSlotNotFoundError();
+
+impl Display for SignatureSlotNotFoundError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Signature slot not found!")
+    }
+}
+
+impl Error for SignatureSlotNotFoundError {}
+
+#[derive(Debug)]
+pub struct MissSyncAggregationError();
+
+impl Display for MissSyncAggregationError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Sync Aggregation not found. \
+        Beacon Block Body in this blockchain variant doesn't contain sync aggregation. \
+        Please use Altair or The Merge variants")
+    }
+}
+
+impl Error for MissSyncAggregationError {}
+
+#[derive(Debug)]
+pub struct ExecutionPayloadError();
+
+impl Display for ExecutionPayloadError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Execution Payload not found. \
+        Beacon Block Body in this blockchain variant doesn't contain execution payload. \
+        Please use The Merge variants")
+    }
+}
+
+impl Error for ExecutionPayloadError {}
+
+#[derive(Debug)]
+pub struct MissSyncCommitteeUpdate();
+
+impl Display for MissSyncCommitteeUpdate {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Sync Committee Update is missing for this Light Client Update")
+    }
 }
 
-/// `BeaconRPCClient` allows getting beacon block body and beacon block header
+impl Error for MissSyncCommitteeUpdate {}
+
+/// `BeaconRPCClient` allows getting beacon block body, beacon block header
+/// and light client updates
 /// using Beacon RPC API (https://ethereum.github.io/beacon-APIs/)
 pub struct BeaconRPCClient {
-    endpoint_url: std::string::String,
+    endpoint_url: String,
+    client: Client,
 }
 
 impl BeaconRPCClient {
     const URL_HEADER_PATH: &'static str = "eth/v1/beacon/headers";
     const URL_BODY_PATH: &'static str = "eth/v2/beacon/blocks";
+    const GET_LIGHT_CLIENT_UPDATE_API: &'static str = "eth/v1/light_client/updates";
```

Comment:
There should be consistency in the naming.

```suggestion
    const URL_GET_LIGHT_CLIENT_UPDATE_API: &'static str = "eth/v1/light_client/updates";
```